lib/logutils.py

```diff
@@ -29,38 +30,57 @@
 class Filter(logging.Filter):
     """
     This class builds a filter to be used in logging.yaml to configure logging
-    
+    name: name of the logger object (regex)
+    module: logger module (regex)
+    msg: logger message (regex)
+    invert: defines whether the filter suppresses the entries matching the above parameters or suppresses everything else (bool)
+    returns: bool
+
     Returning True tells logging to suppress this logentry,
     whereas False will include the record into further processing and eventual output
     """
-    def __init__(self, name=''):
-        if isinstance(name, list):
-            self.name = name
-        elif len(name) == 0:
-            self.name = []
-        else:
-            self.name = [name]
+    def __init__(self, name='', module='', msg='', invert=False):
+        self.name = name if isinstance(name, list) else [] if len(name) == 0 else [name]
+        self.module = module if isinstance(module, list) else [] if len(module) == 0 else [module]
+        self.msg = msg if isinstance(msg, list) else [] if len(msg) == 0 else [msg]
+        self.invert = invert
 
     def filter(self, record):
+        hits = 0
+        total = 3
+        hits = hits + 1 if len(self.name) == 0 else hits
+        hits = hits + 1 if len(self.module) == 0 else hits
+        hits = hits + 1 if len(self.msg) == 0 else hits
         for n in self.name:
-            if n == record.name:
-                return False
-        return True
+            n = re.compile(n)
+            if n.match(record.name):
+                hits += 1
+        for m in self.module:
+            m = re.compile(m)
+            if m.match(record.module):
+                hits += 1
+        for msg in self.msg:
+            msg = re.compile(msg)
+            if msg.match(record.msg):
+                hits += 1
```

Comment:
Just an idea to improve performance: compile regex in `__init__` method instead for each log record.

This could also check if the regex is valid and in case it's invalid it logs an error an will ignore this filter pattern when filtering the logs.

Reply:
Good idea. New commit should do that. Any other tweaks are welcome.